packages/aws/src/lib/CostAndUsageReportsRow.ts

```diff
@@ -22,34 +22,40 @@ import { concat } from 'ramda'
 import { AWS_REPLICATION_FACTORS_FOR_SERVICES } from './ReplicationFactors'
 
 const GLUE_VCPUS_PER_USAGE = 4
-const SIMPLE_DB_VCPUS__PER_USAGE = 1
+const SIMPLE_DB_VCPUS_PER_USAGE = 1
 
 export default class CostAndUsageReportsRow extends BillingDataRow {
-  constructor(usageRowsHeader: Athena.Row, rowData: Athena.datumList) {
-    const billingDataRowKeys = usageRowsHeader.Data.map((column) =>
-      Object.values(column),
-    ).flat()
-    const billingDataRowValues = rowData
-      .map((column) => Object.values(column))
-      .flat()
-    const billingDataRow = Object.fromEntries(
-      billingDataRowKeys.map((_, i) => [
-        billingDataRowKeys[i],
-        billingDataRowValues[i],
-      ]),
+  constructor(rowData: Athena.datumList, tagNames: string[]) {
```

Comment:
The main thing that motivated this change is that Athena returns each row (`rowData` here) as an array like this:

```json
[
  { "VarCharValue": "2022-09-15T00:00:00.0000Z" },
  { "VarCharValue": "my-account" },
  { "VarCharValue": "EC2" },
  { "VarCharValue": "us-east-1" },
  { "VarCharValue": "xxx" },
  { "VarCharValue": "seconds" },
  { "VarCharValue": "2.0" },
  { "VarCharValue": "Production" },
  { "VarCharValue": "3.0" },
  { "VarCharValue": "5.0" }
]
```

If the constructor was to use the original mechanism (turning the row into an object with properties matching the column names in the Athena result), then handling tags would be a little awkward: the tags would get applied to the object with property names like `resource_tags_user_environment`, and then we'd have to do some extra work to pull them out of either the row or the object to put them in `tags`.
  
Directly taking them from the row seemed easier and clearer rather than continuing with the original mechanism.

  
Very open to feedback or other suggestions though! Another option could be to pass in the property values to the constructor. Then all of the knowledge of Athena and the query structure could live in `CostAndUsageReports`, and this class becomes much simpler and has no knowledge of Athena. This would also make the LUT handling easier: we wouldn't need to convert the LUT data into an Athena-like format just for `CostAndUsageReportsRow` to then transform it again.

Reply:
hey Charles, thanks for the explanation. Would it be possible to refactor to pass through the properties to the constructor. I think this direction might be a bit cleaner. The logic for athena could even be extracted out into its own file that is imported to the CUR file. And some of the functions mentioned above could also live there.

Reply:
Resolved in https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/pull/904/commits/f98ca8cafc47a51148ab5974a44bc84cd3ec2058